Update strong naming guidance for .NET libraries to only apply to applicable platforms

baronfel opened the pull request:
Since Strong Naming has no impact on modern .NET Runtimes, limit the scope of guidance to .NET Framework and .NET Standard


<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/library-guidance/strong-naming.md](https://github.com/dotnet/docs/blob/9efc8978441d1c59634197bfcd68c3848dd7ee77/docs/standard/library-guidance/strong-naming.md) | [Strong naming](https://review.learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming?branch=pr-en-us-49026) |


<!-- PREVIEW-TABLE-END -->

baronfel committed:
Update strong naming guidance for .NET libraries to only apply to applicable platforms

Since Strong Naming has no impact on modern .NET Runtimes, limit the scope of guidance to .NET Framework and .NET Standard

gewarren reviewed (approved):
Left a suggestion for consideration. Thanks for adding this!

baronfel committed:
Update docs/standard/library-guidance/strong-naming.md

Co-authored-by: Genevieve Warren <24882762+gewarren@users.noreply.github.com>

baronfel committed:
clarify multi-targeting guidelines

If multitargeting, strong naming should be done for all TFMs so that InternalsVisibleTo attributes can remain consistent.

baronfel committed:
fix spacing

IEvangelist committed:
Update strong-naming.md
